powerlibs/django/restless/contrib/endpoints/base.py

```diff
@@ -71,6 +72,22 @@ def get_query_set(self, request, *args, **kwargs):
 
 
 class SoftDeletableDetailEndpointMixin:
+
+    def serialize(self, data):
+        return serialize(data, exclude=['deleted'])
+
+    def put(self, request, *args, **kwargs):
+        if 'deleted' in request.data:
+            return Http400({"error": "Invalid field name: deleted"})
+
+        return super().put(request, *args, **kwargs)
+
+    def patch(self, request, *args, **kwargs):
+        if 'deleted' in request.data:
```

Comment:
Can you see you're repeating that a lot? And that this code is absolutely independent from which method is being called?
A nice **decorator** would be a much cleaner implementation.